### PR TITLE
Handle broken imports and dependencies from tagging

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes: https://github.com/dart-lang/pana/issues/675

### CODE REVIEW

File: lib/src/tag_detection.dart [hunk 1]
```diff
@@ -824,125 +830,154 @@ class Tagger {
...
-        // Report only the first non-pruned violation as Explanation
-        final firstNonPrunedViolation = _topLibraries
-            .map(violationFinder._findPlatformViolation)
-            .firstWhere((e) => e != null, orElse: () => null);
-        if (firstNonPrunedViolation != null) {
-          explanations.add(firstNonPrunedViolation);
-        }
+    try {
+      if (_isBinaryOnly) {
+        Explanation('Binary only',
```
Comment (isoos) [inline lib/src/tag_detection.dart:835]: shouldn't this be `explanations.add(...)`?

File: lib/src/tag_detection.dart [hunk 2]
```diff
@@ -824,125 +830,154 @@ class Tagger {
...
+            'Cannot assign flutter platform tags, it is a binary only package');
+      } else if (_topLibraries.isEmpty) {
+        Explanation('No top-level libraries found',
```
Comment (isoos) [inline lib/src/tag_detection.dart:838]: shouldn't this be `explanations.add(...)`?

File: lib/src/tag_detection.dart [hunk 3]
```diff
@@ -824,125 +830,154 @@ class Tagger {
...
-      final dartSdkViolationFinder = _SdkViolationFinder(
-          _packageGraph, _Sdk.dart, _pubspecCache, _session);
-      if (dartSdkViolationFinder.findSdkViolation(packageName, _topLibraries) !=
-          null) {
-        // This is reported elsewhere
+    try {
+      if (_isBinaryOnly) {
+        tags.addAll(<String>[Runtime.nativeAot.name, Runtime.nativeJit.name]);
+      } else if (_topLibraries.isEmpty) {
+        Explanation('No top-level libraries found',
```
Comment (isoos) [inline lib/src/tag_detection.dart:914]: shouldn't this be `explanations.add(...)`?

File: lib/src/tag_detection.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/tag_detection.dart:838]: yes!!

File: lib/src/tag_detection.dart (hunk 3 above)
Comment (sigurdm) [reply lib/src/tag_detection.dart:914]: yes!!

**Pull request merged**